### PR TITLE
update jackson checksums

### LINKED ISSUE #95: Build fails because downloading jackson-module-jaxb-annotations dependency gives a 404 error
`https://mvnrepository.com/com/fasterxml/jackson/module/jackson-module-jaxb-annotations/maven-metadata.xml` seems to not exist anymore.

`./gradlew build --scan`:

```
> Configure project :desktop
Evaluating project ':desktop' using build file '/haveno/desktop/build.gradle'.
Failed to get resource: GET. [HTTP HTTP/1.1 403 Forbidden: https://mvnrepository.com/com/fasterxml/jackson/module/jackson-module-jaxb-annotations/maven-metadata.xml)]
Verifying aopalliance:aopalliance
Verifying ch.qos.logback:logback-classic
Verifying ch.qos.logback:logback-core
Verifying com.fasterxml.jackson.core:jackson-annotations

FAILURE: Build failed with an exception.
```

### PR DESCRIPTION
**Pull request by woodser**
Fixes #95

### CODE REVIEW

**Review by erciccione — APPROVED** [review]

**Comment by premek** [comment]
interesting, why did the checksums of the same version changed, were they different on different repositories and taken down from one?

**Comment by woodser** [comment]
> interesting, why did the checksums of the same version changed, were they different on different repositories and taken down from one?

I don't know. These checksums were automatically provided by removing the non-working checksums then running `./gradlew -q calculateChecksums`.

We should confirm they are valid. Any ideas?

**Comment by premek** [comment]
Looks like monero-java depends on jackson 2.8.11.1 *or newer*:
- https://github.com/monero-ecosystem/monero-java/blob/master/pom.xml
- https://www.mojohaus.org/versions-maven-plugin/examples/resolve-ranges.html

We specify `jacksonVersion = '2.12.1'` but through the transitive dependency we get 2.12.4 which was released few days ago. 

The checksum matches:
```
$ curl -s https://repo1.maven.org/maven2/com/fasterxml/jackson/core/jackson-annotations/2.12.4/jackson-annotations-2.12.4.jar | sha256sum 
f6aa3706a875689b66cdac3334f65dfdb795ccfad4117bf072893b196ed1ec8e  -
```

So I still don't know how to verify the checksum but this seems to be the reason why do we need to update it (and will have to when they release a new versions in future)

**Review by premek — APPROVED** [review]

**Comment by woodser** [comment]
I'm seeing that these values have changed again.

**Comment by premek** [comment]
because they released 2.13.0-rc1?

**Comment by woodser** [comment]
> because they released 2.13.0-rc1?

Sounds likely. We need a way to specify the version without getting bumped up through transitive dependencies.

File: build.gradle [hunk 1]
```diff
@@ -407,6 +412,13 @@ configure(project(':desktop')) {
         preserveFileTimestamps = false
         reproducibleFileOrder = true
     }
+    
+    configurations.all {
+        resolutionStrategy {
+            failOnChangingVersions()
+            failOnNonReproducibleResolution()
+        }
+    }
```
Comment (premek) [inline build.gradle:421]: this is good 
but does not lock the transitive dependency version, right?

File: build.gradle [hunk 2]
```diff
@@ -321,6 +321,11 @@ configure(project(':core')) {
             exclude(module: 'base64')
             exclude(module: 'httpcore-nio')
         }
+        compile "com.fasterxml.jackson.core:jackson-core:$jacksonVersion"
+        compile "com.fasterxml.jackson.core:jackson-annotations:$jacksonVersion"
+        compile("com.fasterxml.jackson.core:jackson-databind:$jacksonVersion") {
+            exclude(module: 'jackson-annotations')
+        }
```
Comment (premek) [inline build.gradle:328]: isn't this duplicated on lines 336-340 now?

**Pull request closed without merging**